src/main/scala/subsystem/BaseSubsystem.scala

```diff
@@ -19,22 +19,29 @@ case object InTile extends HierarchicalLocation("InTile")
 case object InSubsystem extends HierarchicalLocation("InSubsystem")
 case object InSystem extends HierarchicalLocation("InSystem")
 
-/** BareSubsystem is the root class for creating a subsystem */
-abstract class BareSubsystem(implicit p: Parameters) extends LazyModule with BindingScope {
+// HasDts is generating metadatas from Scala, which is not the target for new diplomacy and Property.
+// It will be deprecated and removed after we migrate all metadata handling logic to OM Dialect.
+trait HasDts extends LazyModule with BindingScope {
```

Comment:
Nit: should be `HasDTS`

Reply:
FWIW: camel casing like this is a common style suggestion: 
 https://google.github.io/styleguide/javaguide.html 5.3